int16 n
---
string direction
